src/EditorFeatures/CSharpTest/SymbolKey/SymbolKeyCompilationsTests.cs

```diff
@@ -239,6 +239,94 @@ public void M<Z>(List<Z> list)
             ResolveAndVerifySymbolList(members1, members2, comp1);
         }
 
+        [Fact]
+        public void FileType1()
+        {
+            var src1 = @"using System;
+
+namespace N1.N2
+{
+    file class C { }
+}
+";
+            var originalComp = CreateCompilation(src1, assemblyName: "Test");
+            var newComp = CreateCompilation(src1, assemblyName: "Test");
+
+            var originalSymbols = GetSourceSymbols(originalComp, SymbolCategory.DeclaredType | SymbolCategory.DeclaredNamespace).OrderBy(s => s.Name);
+            var newSymbols = GetSourceSymbols(newComp, SymbolCategory.DeclaredType | SymbolCategory.DeclaredNamespace).OrderBy(s => s.Name);
+
+            ResolveAndVerifySymbolList(newSymbols, originalSymbols, originalComp);
+        }
+
+        [Fact]
+        public void FileType2()
+        {
+            var src1 = @"using System;
+
+namespace N1.N2
+{
+    file class C<T> { }
+}
+";
+            var originalComp = CreateCompilation(src1, assemblyName: "Test");
+            var newComp = CreateCompilation(src1, assemblyName: "Test");
+
+            var originalSymbols = GetSourceSymbols(originalComp, SymbolCategory.DeclaredType | SymbolCategory.DeclaredNamespace).OrderBy(s => s.Name);
+            var newSymbols = GetSourceSymbols(newComp, SymbolCategory.DeclaredType | SymbolCategory.DeclaredNamespace).OrderBy(s => s.Name);
+
+            ResolveAndVerifySymbolList(newSymbols, originalSymbols, originalComp);
+        }
+
+        [Fact]
+        public void FileType3()
+        {
+            var src1 = @"using System;
+
+namespace N1.N2
+{
+    file class C { }
+}
+";
+            // this should result in two entirely separate file symbols.
+            // note that the IDE can only distinguish file type symbols with the same name when they have distinct file paths.
```

Comment:
>note that the IDE can only distinguish file type symbols [](http://example.com/codeflow?start=15&length=56)

Are we asserting that we found distinct file type symbols? Should we include `Assert.Equal(3, originalSymbols.Length);`?